go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -506,7 +506,22 @@ func (qre *QueryExecutor) execDDL(conn *StatefulConnection) (*sqltypes.Result, e
 	}
 
 	defer func() {
-		if err := qre.tsv.se.Reload(qre.ctx); err != nil {
+		var created, altered, dropped []string
+
+		switch ddlStmt := qre.plan.FullStmt.(type) {
+		case *sqlparser.CreateTable:
+			created = append(created, ddlStmt.Table.Name.String())
+
+		case *sqlparser.AlterTable:
+			altered = append(altered, ddlStmt.Table.Name.String())
+
+		case *sqlparser.DropTable:
+			for _, tableName := range ddlStmt.FromTables {
+				dropped = append(dropped, tableName.Name.String())
+			}
+		}
+
```

Comment:
I think these are not enough, have to look at all the DDL Statement types
check `buildDDLPlans` function in `go/vt/vtgate/planbuilder/ddl.go`

Reply:
okay! thanks!!1 ✨

Reply:
Indeed, looking at [go/vt/vtgate/planbuilder/ddl.go:109-139](https://github.com/pudiva/vitess/blob/4769de3b550c6e28aed8fc218247560df6a129f1/go/vt/vtgate/planbuilder/ddl.go?plain=1#L109-L139), we could cover more cases with this optimisation.

At the moment, if the DDL is not picked up by the switch statement, it will just skip the optimisation and reload everything.

I'll come back to this after I sort out the segfault on CI.